core/jazz_events-handler/test/test.js

```diff
@@ -1,10 +1,263 @@
+        return cb(err, null);
+      });
+      index.processEachEvent(event.Records[0], config, interestedEvents)
+        .catch(error => {
+          expect(error).to.deep.eq(err);
+          AWS.restore("DynamoDB.DocumentClient");
+        });
+    });
 
-describe('Sample', function() {
-  it('tests handler', function(done) {
+    it("should process interested event", () => {
+      AWS.mock("DynamoDB.DocumentClient", "put", (params, cb) => {
+        let data = {
+          "message": "success"
+        }
+        return cb(null, data);
+      });
+      let message = {
+        "message": "successfully processed interested event"
+      };
 
-    // Add your test cases here.
-    assert(true);
-    done();
+      index.processEachEvent(event.Records[0], config, interestedEvents)
+        .then(res => {
+          expect(res).to.deep.eq(message);
+          AWS.restore("DynamoDB.DocumentClient");
+        });
+    });
+
+    it("should not process un-interested event", () => {
+      event.Records[0].kinesis.partitionKey = "DELETE_ASSET";
+      let message = {
+        "message": "successfully processed un-interested event"
+      };
+
+      index.processEachEvent(event.Records[0], config, interestedEvents)
+        .then(res => {
+          expect(res).to.deep.eq(message);
+        });
+    });
   });
+
+  describe('processRecords tests', function () {
+    beforeEach(() => {
+      interestedEvents = [
+        'CREATE_DEPLOYMENT',
+        'CREATE_ASSET',
+        'UPDATE_DEPLOYMENT',
+        'CREATE_BRANCH',
+        'UPDATE_BRANCH',
+        'DELETE_BRANCH'
+      ]
+    });
+
+    it("should not indicate an error when DynamoDB.DocumentClient.put throws an error", () => {
+      AWS.mock("DynamoDB.DocumentClient", "put", (params, cb) => {
+        return cb(err, null);
+      });
+      index.processRecords(config, event, interestedEvents)
+        .then(res => {
+          expect(res).to.be.undefined;
+          AWS.restore("DynamoDB.DocumentClient");
+        })
+    });
+
+    it("should successfully process interested events", () => {
+      let result = [
+        { "message": "successfully processed interested event" },
+        { "message": "successfully processed interested event" }
+      ];
+      const processEachEvent = sinon.stub(index, "processEachEvent")
+        .resolves({ "message": "successfully processed interested event" });
+
+      index.processRecords(config, event, interestedEvents)
+        .then(res => {
+          expect(res).to.deep.eq(result);
+          sinon.assert.calledTwice(processEachEvent)
+          processEachEvent.restore();
+        });
+    });
+  });
+
+  describe('handler', function() {
+    it("should successfully process event", () => {
+      const getEvents = sinon.stub(index, "getEvents").resolves(interestedEvents);
+      const processRecords = sinon.stub(index, "processRecords").resolves();
+
+      index.handler(event, context, (error, data) => {
```

Comment:
dont u have to verify some data?

Reply:
Done
Intially had left it, as i was verifying the data in parts in individual functions. On taking another look its better to verify the data again.